python/lsst/sphgeom/unitVector3d.cc

```diff
@@ -117,16 +117,14 @@ PYBIND11_MODULE(unitVector3d, mod) {
     // components. Furthermore, UnitVector3d::fromNormalized is not visible to
     // Python, and even if it were, pybind11 is currently incapable of returning
     // a picklable reference to it.
-    cls.def("__getstate__", [](UnitVector3d const &self) {
-        return py::make_tuple(self.x(), self.y(), self.z());
-    });
-    cls.def("__setstate__", [](UnitVector3d &self, py::tuple t) {
-        if (t.size() != 3) {
-            throw std::runtime_error("Invalid state!");
-        }
-        new (&self) UnitVector3d(UnitVector3d::fromNormalized(
-                t[0].cast<double>(), t[1].cast<double>(), t[2].cast<double>()));
-    });
+    cls.def(py::pickle([](UnitVector3d const &self) { return py::make_tuple(self.x(), self.y(), self.z()); },
+                       [](py::tuple t) {
+                           if (t.size() != 3) {
+                               throw std::runtime_error("Invalid state!");
+                           }
+                           return new UnitVector3d(UnitVector3d::fromNormalized(
+                                   t[0].cast<double>(), t[1].cast<double>(), t[2].cast<double>()));
```

Comment:
I know this was already there, but why a heap one?

Reply:
I'm not sure what you are asking. Are you asking why this code returns a pointer instead of a vector? If so, my guess is because `UnitVector3d` has no move constructor -- though I don't know if the compiler would still manage to move the data in this situation.